app/models.py

```diff
@@ -18,10 +18,23 @@
 
 
 class PieceMove:
-    from_x: int
-    from_y: int
-    to_x: int
-    to_y: int
+    from_pos: tuple
+    to_pos: tuple
+
+    def __init__(self, board_move_split):
+        from_board_pos, to_board_pos = (board_move_split[0], board_move_split[1])
+        self.from_pos = PieceMove.position_converter(from_board_pos)
+        self.to_pos = PieceMove.position_converter(to_board_pos)
+
+    @staticmethod
+    def position_converter(board_pos: str) -> tuple[int, int]:
+        board_pos_split = [*board_pos]
+        x_pos, y_pos = (board_pos_split[0].upper(), board_pos_split[1])
```

Comment:
You can also remove `board_pos_split` and call each character from a string directly:
```python
my_str = "abc"
print(my_str[0]) # 'a'
print(my_str[1]) # 'b'
print(my_str[2]) # 'c'
```

So, this could also be:
```suggestion
        x_pos, y_pos = (board_pos[0].upper(), board_pos[1])
```

Reply:
correct! nice one